src/MahApps.Metro/Controls/Dialogs/MessageDialog.cs

```diff
@@ -131,7 +135,32 @@ public MessageDialogStyle ButtonStyle
             set => this.SetValue(ButtonStyleProperty, value);
         }
 
-        #endregion Properties
+        /// <summary>Identifies the <see cref="Icon"/> dependency property.</summary>
+        public static readonly DependencyProperty IconProperty
+            = DependencyProperty.Register(nameof(Icon),
+                                          typeof(object),
+                                          typeof(MessageDialog),
+                                          new PropertyMetadata());
+
+        public object? Icon
+        {
+            get => this.GetValue(IconProperty);
+            set => this.SetValue(IconProperty, value);
+        }
+
+        /// <summary>Identifies the <see cref="IconTemplate"/> dependency property.</summary>
+        public static readonly DependencyProperty IconTemplateProperty
+            = DependencyProperty.Register(nameof(IconTemplate),
+                                          typeof(DataTemplate),
+                                          typeof(MessageDialog));
+
+        public DataTemplate? IconTemplate
+        {
+            get => (DataTemplate?)this.GetValue(IconTemplateProperty);
+            set => this.SetValue(IconTemplateProperty, value);
+        }
```

Comment:
Is it not better to move this to the base class? Because it is used in each derived class. So a user can use this in a custom dialog too. WDYT?